### ISSUE TITLE
Implement Rust symbol demanling

### ISSUE THREAD

**Issue by stanislaw** [mull-project/mull#122]
Needed to make reports more readable so that

```html
_ZN8demangle5tests8demangle17h4d833f55bb2fea15E
_ZN8demangle5tests16demangle_dollars17h654ff4516cb2a88eE
_ZN8demangle5tests21demangle_many_dollars17h929844db6571e81cE
```

becomes:

```txt
demangle::tests::demangle
demangle::tests::demangle_dollars
demangle::tests::demangle_many_dollars
```

Direct port from Rust to C++ should work:

https://github.com/alexcrichton/rustc-demangle.

**Comment by AlexDenisov** [comment]
What's the purpose? Where do you want to use it?

**Comment by stanislaw** [comment]
See [here](https://github.com/stanislaw/Talks/blob/master/2017.02.23-LLVM-Social-Berlin-6-Mull-meets-Rust/index.html#L482) :)

**Comment by stanislaw** [comment]
Closed by #136 .
